contracts/command-batcher/src/encoding/evm.rs

```diff
@@ -0,0 +1,483 @@
+use std::{collections::HashMap, str::FromStr};
+
+use axelar_wasm_std::Snapshot;
+use connection_router::msg::Message;
+use cosmwasm_std::{HexBinary, Uint256};
+use ethabi::{ethereum_types, Token};
+use k256::{elliptic_curve::sec1::ToEncodedPoint, PublicKey};
+use multisig::types::Signature;
+use sha3::{Digest, Keccak256};
+
+use crate::{
+    error::ContractError,
+    types::{Command, CommandBatch, CommandType, Data, Operator, Proof},
+};
+
+use super::traits;
+
+impl TryFrom<Message> for Command {
+    type Error = ContractError;
+
+    fn try_from(msg: Message) -> Result<Self, Self::Error> {
+        Ok(Command {
+            command_type: CommandType::ApproveContractCall, // TODO: this would change when other command types are supported
+            command_params: command_params(
+                msg.source_chain,
+                msg.source_address,
+                ethereum_types::Address::from_str(&msg.destination_address).map_err(|_| {
+                    ContractError::InvalidMessage {
+                        reason: "destination_address is not a valid EVM address".into(),
+                    }
+                })?,
+                msg.payload_hash.as_slice().try_into().map_err(|_| {
+                    ContractError::InvalidMessage {
+                        reason: "payload_hash is not a valid keccak256 hash".into(),
+                    }
+                })?,
+            ),
+            id: command_id(msg.id),
+        })
+    }
+}
+
+pub struct Builder;
+impl traits::Builder for Builder {
+    fn build_batch(
+        messages: Vec<Message>,
+        destination_chain_id: Uint256,
+    ) -> Result<CommandBatch, ContractError> {
+        let message_ids: Vec<String> = messages.iter().map(|msg| msg.id.clone()).collect();
```

Comment:
We should probably sort these message ids, so that way the batch id is not affected by the order of message ids

Reply:
I did the sorting inside the batch_id function itself, I didn't want to do it from here, because it would also change the order of other stuff that also depends on message_ids, like the command vector, but I would assume that it should respect the order in which the messages were relayed